privacy_analytics/managers.py

```diff
@@ -0,0 +1,25 @@
+from django.db.models import Manager, Avg, Count, Q, FloatField
+from django.db.models.functions import Cast
+
+
+class PageViewManager(Manager):
+    def unique(self, **kwargs):
+        return self.filter(**kwargs).values('session_key').distinct().count()
+
+    def views_per_visitor(self, **kwargs):
+        return self.filter(**kwargs).values('session_key').annotate(total=Count('pk')).aggregate(Avg('total'))
+
+    def total_views(self, **kwargs):
+        return self.filter(**kwargs).values('session_key').count()
+
+    def percent_authenticated(self, **kwargs):
+        return self.filter(**kwargs).aggregate(total=(Cast(Count('is_authenticated', filter=Q(is_authenticated=True)), FloatField()) / Cast(Count('pk'), FloatField())) * 100.00)
```

Comment:
What happens when the data is empty? Wouldn't this error due to dividing by zero, or does the database handle this differently?

Reply:
I have updated all of the manager logic to combat this and this was a janky fix